tutorials/pretrain_redpajama.md

```diff
@@ -0,0 +1,173 @@
+```
+
+```bash
+# The 1 billion token subset
+# Make sure you have git-lfs installed (https://git-lfs.com): git lfs install
+git clone https://huggingface.co/datasets/togethercomputer/RedPajama-Data-1T-Sample data/RedPajama-Data-1T-Sample
+```
+
+## Prepare RedPajama for training
+
+The full dataset consists of 2084 `jsonl` files (the sample dataset contains 11). In order to start pretraining lit-gpt
+on it, you need to read, tokenize, and write the data in binary chunks. This will leverage the `PackedDataset`
+streaming dataset that comes with lit-gpt.
+
+Do to so, run
+
+```bash
+python scripts/prepare_redpajama.py --source_path data/RedPajama-Data-1T --checkpoint_dir checkpoints/meta-llama/Llama-2-7b-hf/ --destination_path data/lit-redpajama
+```
+
+or
+
+```bash
+python scripts/prepare_redpajama.py --source_path data/RedPajama-Data-1T-Sample --checkpoint_dir checkpoints/meta-llama/Llama-2-7b-hf/ --destination_path data/lit-redpajama-sample --sample True
+```
+
+for the sample dataset.
+
+In the above we are assuming that you will be using the same tokenizer as used in LLaMA, but any trained [SentencePiece](https://github.com/google/sentencepiece) tokenizer with a 32000 vocabulary size will do here.
+
+The script will take a while to run, so time for :tea:. (The 1B sample script takes about 45 min for the data preparation.)
+
+## Pretraining
+
+Running the pretraining script with its default settings requires at least 4 GPUs with 40GB+ each (A100).
+
+```bash
+python pretrain/redpajama.py --devices 4 --train_data_dir data/lit-redpajama
+```
+
+For running on the sample dataset:
+
+```bash
+python pretrain/redpajama.py --devices 4 --train_data_dir data/lit-redpajama-sample
+```
+
+The script will save checkpoints periodically to the folder `out/`.
+
+By default, the `train_redpajama.py` script will pretrain the Llama 2 7B model with FSDP in
+`bfloat16` precision and gradient accumulation.
+
+You can easily change the size of the model by passing a different string to the model name variable
+
+```python
+model_name = "Llama-2-7b-hf"
+```
+
+at the top of this script.
+
+
+The currently supported model names are contained in the [config.py](https://github.com/Lightning-AI/lit-gpt/lit_gpt/config.py#L114) file and include:
+
+
+- falcon-7b
+- falcon-40b
+- FreeWilly2
+- Llama-2-7b-hf
+- Llama-2-13b-hf
+- Llama-2-70b-hf
+- longchat-7b-16k
+- longchat-13b-16k
+- Nous-Hermes-13b
+- open_llama_3b
+- open_llama_7b
+- open_llama_13b
+- pythia-70m
+- pythia-160m
+- pythia-410m
+- pythia-1b
+- pythia-1.4b
+- pythia-2.8b
+- pythia-6.9b
+- pythia-12b
+- RedPajama-INCITE-{}-3B-v1
+- RedPajama-INCITE-7B-{}
+- RedPajama-INCITE-{}-7B-v0.1
+- stablelm-base-alpha-3b
+- stablelm-base-alpha-7b
+- stablelm-tuned-alpha-3b
+- stablelm-tuned-alpha-7b
+- vicuna-7b-v1.3
+- vicuna-13b-v1.3
+- vicuna-33b-v1.3
+- vicuna-7b-v1.5
+- vicuna-7b-v1.5-16k
+- vicuna-13b-v1.5
+- vicuna-13b-v1.5-16k
```

Comment:
Instead of listing this explicitly, I would link to our config.py file

Reply:
agree, we can refer to the config file and also suggest a programmatic approach: 

```
from lit_gpt.config import configs

for conf in configs:
    print(conf["name"])
```